lib/quickbooks/service/access_token.rb

```diff
@@ -15,24 +17,30 @@ def renew
             result = Quickbooks::Model::AccessTokenResponse.from_xml(response.plain_body)
           end
         end
-
         result
+
       end
 
       # https://developer.intuit.com/docs/0025_quickbooksapi/0053_auth_auth/oauth_management_api#Disconnect
       def disconnect
-        result = nil
-        response = do_http_get(DISCONNECT_URL)
-        if response
-          code = response.code.to_i
-          if code == 200
-            result = Quickbooks::Model::AccessTokenResponse.from_xml(response.plain_body)
+        if oauth_v1?
+          response = do_http_get(DISCONNECT_URL_OAUTH1)
+          if response && response.code.to_i == 200
+            Quickbooks::Model::AccessTokenResponse.from_xml(response.plain_body)
+          end
+        elsif oauth_v2?
+          conn = Faraday.new
+          conn.basic_auth oauth.client.id, oauth.client.secret
+          response = conn.post(DISCONNECT_URL_OAUTH2, token: oauth.refresh_token || oauth.token)
+          if response.success?
+            Quickbooks::Model::AccessTokenResponse.new(error_code: "0")
+          else
+            Quickbooks::Model::AccessTokenResponse.new(
+              error_code: response.status.to_s, error_message: response.body
+            )
```

Comment:
response.reason_phrase give error "undefined method `reason_phrase' for #<Faraday::Response:0x0055e03cd06c78>".

Reply:
We need to add this PR too https://github.com/ruckus/quickbooks-ruby/pull/470/files